extensions/amp-story/1.0/test/test-amp-story.js

```diff
@@ -1325,6 +1333,424 @@ describes.realWin(
       });
     });
 
+    describe('amp-story-subscriptions navigation', () => {
+      let subscriptionsEl;
+      let storeService;
+
+      const clickRightEvent = new MouseEvent('click', {clientX: 200});
+      const clickLeftEvent = new MouseEvent('click', {clientX: 10});
+
+      async function setUpStorySubscriptions() {
+        await createStoryWithPages(4, ['cover', 'page-1', 'page-2', 'page-3']);
+        subscriptionsEl = win.document.createElement('amp-story-subscriptions');
+        story.element.appendChild(subscriptionsEl);
+
+        // buildCallback() is implicitly called by createStoryWithPages()
+        await story.layoutCallback();
+
+        const cover = story.getPageById(
+          storeService.get(StateProperty.CURRENT_PAGE_ID)
+        );
+        cover.element.dispatchEvent(clickRightEvent);
+      }
+
+      beforeEach(async () => {
+        toggleExperiment(win, 'amp-story-subscriptions', true);
+
+        storeService = new AmpStoryStoreService(win);
+        env.sandbox.stub(Services, 'storyStoreService').returns(storeService);
+
+        // This stub makes requestAnimationFrame a sync call so that each dispatch of click event
+        // can be a sync operation, which means it can be used as doing await switchTo call.
+        env.sandbox.stub(win, 'requestAnimationFrame').callsFake((cb) => cb());
+      });
+
+      describe('UNKNOWN subscription state before paywall page', () => {
+        beforeEach(async () => {
+          await setUpStorySubscriptions();
+        });
+
+        it('should not navigate to subscription content while subscription state is unknown', async () => {
+          const page1 = story.getPageById(
+            storeService.get(StateProperty.CURRENT_PAGE_ID)
+          );
+          expect(page1.element.id).to.equal('page-1');
+
+          // Attempt to go to page-2 but should stay on page-1 since the state is unknown.
+          page1.element.dispatchEvent(clickRightEvent);
+          await nextTick();
+          const activePage = story.getPageById(
+            storeService.get(StateProperty.CURRENT_PAGE_ID)
+          );
+          expect(activePage.element.id).to.equal('page-1');
+        });
+
+        it('should continue normal navigation after the subscription state gets resolved to granted', async () => {
+          const page1 = story.getPageById(
+            storeService.get(StateProperty.CURRENT_PAGE_ID)
+          );
+          page1.element.dispatchEvent(clickRightEvent);
+          // Once grant status resolves to true, goes back to normal story experience.
+          storeService.dispatch(
```

Comment:
Please correct me if I'm wrong, but it's navigation to page-3 that is blocked by your paywall code in `switchTo_()`, so this test doesn't work?